lib/factory_bot.rb

```diff
@@ -101,6 +104,7 @@ class << self
     deprecate :allow_class_lookup,
               :allow_class_lookup=,
               :factory_by_name,
+              :register_callback,
```

Comment:
Calling ```register_strategy``` or ```register_callback``` interacts directly with it, and we get a lot of deprecation warnings. Maybe we can but I think it's better to do it in a separate PR, because we need to change the strategy delegating those.